backend/functions/lib/adapters/gateways/case.assignment.local.repository.test.ts

```diff
@@ -1,23 +1,133 @@
+      assignedOn: currentDate,
+    };
+
+    // First create a record to update.
+    await localRepo.createAssignment(assignementToCreate);
+    const created = await localRepo.findAssignmentsByCaseId(caseId);
+    expect(created[0]).toEqual(assignementToCreate);
+
+    // Then update the record.
+    const assignmentToUpdate: CaseAttorneyAssignment = {
+      ...assignementToCreate,
+      unassignedOn: new Date().toISOString(),
+    };
+    await localRepo.updateAssignment(assignmentToUpdate);
+    const updated = await localRepo.findAssignmentsByCaseId(caseId);
+    expect(updated[0]).toEqual(assignmentToUpdate);
+  });
+
+  test('Should throw error if case is not found while updating', async () => {
+    const applicationContext = await applicationContextCreator(context);
+    const localRepo: CaseAssignmentRepositoryInterface = new CaseAssignmentLocalRepository(
+      applicationContext,
+    );
+
+    const caseId = '123';
+    const currentDate = new Date().toISOString();
+    const assignment = {
+      caseId,
+      name: 'Susan Arbeit',
+      role: CaseAssignmentRole['TrialAttorney'],
+      assignedOn: currentDate,
+      unassignedOn: new Date().toISOString(),
+    };
+
+    expect(async () => {
+      await localRepo.updateAssignment(assignment);
+    }).rejects.toThrow(
+      new UnknownError('LOCAL-ASSIGNMENT-REPOSITORY', {
+        message:
+          'Unable to update assignment. Please try again later. If the problem persists, please contact USTP support.',
+        originalError: new Error('Can not find record'),
+        status: 500,
+      }),
+    );
+  });
+
+  test('Should return assignment by id', async () => {
+    const applicationContext = await applicationContextCreator(context);
+    const localRepo: CaseAssignmentRepositoryInterface = new CaseAssignmentLocalRepository(
+      applicationContext,
+    );
+
+    const caseId = '123';
+    const currentDate = new Date().toISOString();
+    const assignment = {
+      caseId,
+      name: 'Susan Arbeit',
+      role: CaseAssignmentRole['TrialAttorney'],
+      assignedOn: currentDate,
+    };
+
+    // First create a record to update.
+    await localRepo.createAssignment(assignment);
+    const created = await localRepo.findAssignmentsByCaseId(caseId);
+    expect(created[0]).toEqual(assignment);
+
+    const returned = await localRepo.getAssignment(created[0].id);
+    expect(returned).toEqual(assignment);
+  });
+
+  test('Should return assignment by name', async () => {
+    const applicationContext = await applicationContextCreator(context);
+    const localRepo: CaseAssignmentRepositoryInterface = new CaseAssignmentLocalRepository(
+      applicationContext,
+    );
+
+    const caseId = '123';
+    const currentDate = new Date().toISOString();
+    const assignment = {
+      caseId,
+      name: 'Susan Arbeit',
+      role: CaseAssignmentRole['TrialAttorney'],
+      assignedOn: currentDate,
+    };
+
+    // First create a record to update.
```

Comment:
Looks like we copied this comment from above but it doesn't mean the same thing here.